src/plots/cartesian/axes.js

```diff
@@ -732,6 +731,25 @@ function arrayTicks(ax) {
     return ticksOut;
 }
 
+function roundBaseDay(dayHours) {
+    switch(dayHours) {
+        case 4: return [1, 2];
+        case 6: return [1, 2, 3];
+        case 8: return [1, 2, 4];
+        case 9: return [1, 3];
+        case 10: return [1, 2, 5];
+        case 12: return [1, 2, 3, 4, 6];
+        case 14: return [1, 2, 7];
+        case 15: return [1, 3, 5];
+        case 16: return [1, 2, 4, 8];
+        case 18: return [1, 2, 3, 6, 9];
+        case 20: return [1, 2, 4, 5, 10];
+        case 21: return [1, 3, 7];
+        case 22: return [1, 2, 11];
```

Comment:
Oh interesting... so just a precise list of factors of the rounded number of visible hours. I'm not sure we really want *all* of these, and we might want to add *something* intermediate for the larger primes so we don't jump straight from 1 day to 1 hour, even though we can't do it precisely uniformly. But this is a great starting point anyway, we can tweak later.

However, looking at the mocks changed in this PR it seems like there's a problem when `dtick` is not an even fraction of a day. For example in `axes_breaks-candlestick2`, `dtick` is 18 hours (how did that happen? I only see 9 here, I would have thought it would go to 24 after that? That doesn't matter though, there are lots of periods we want that don't divide 24 evenly, this is just an example), and as a result some days have 2 ticks (at 9:30 and 12:00) and other days have only one (at 9:30).

Trying to figure out a reasonable way to handle this, in conjunction with `tick0` possibly being set manually. I think what we may need to say is: when you have hourly rangebreaks and `dtick` < 24h, the only piece of `tick0` that we consider is the time portion and we start at that time on each new day, then increment by `dtick` until we get to the next day, at which point we reset the time to `tick0` again, so you get ticks at the same hours in every day.